apps/public/public.go

```diff
@@ -108,6 +111,13 @@ func WithHTTPClient(httpClient ops.HTTPClient) Option {
 	}
 }
 
+// WithInstanceDiscovery set to false to disable authority validation (to support private cloud scenarios)
+func WithInstanceDiscovery(enabled bool) Option {
```

Comment:
[TestWithTenantID](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/4abfea36f99928ad52523e017b804616993470f5/apps/confidential/confidential_test.go#L713) is a good starting point. A test for this option would be much simpler. It can just run through the happy path for each AcquireToken method and assert that it got the expected token. If the client doesn't send the expected sequence of requests set on the mock client, you would see an error from the AcquireToken method or a panic.